File: 프로그래머스/0/120871. 저주의 숫자 3/저주의 숫자 3.java
class Solution {
    public int solution(int n) {
        int ret = 0;
        int[] temp = new int[110];
        int k = 0;
        int i = 1;
        while(temp[n+1] == 0){
            k++;
            if(k % 3 == 0 || (""+k).contains("3")) continue;
            temp[i] = k; i++;
            System.out.println(i + " :: " + k);
        }
        ret = temp[n];
        return ret;
    }
}

File: 프로그래머스/0/120871. 저주의 숫자 3/저주의 숫자 3.py
def check(num):
    for c in str(num):
        if c == '3':
            return True
    return num % 3 == 0

def solution(n):
    ret = {}
    num = 1
    for i in range(1, n+1):
        ret[i] = num
        num += 1
        while check(num):
            num += 1
    return ret[n]
# 3배수, 3사용x